src/surfing_penguin/API/survey_api.py

```diff
@@ -118,14 +139,33 @@ def post(self):
         return survey_functions.name_get_survey(api.payload['name'])
 
 
+@api.route('/search_survey_by_author')
+class search_survey_by_author(Resource):
+    """
+    Show the information of a survey, given its name or ID.
+    """
+    @api.marshal_list_with(api_return_survey)
+    @api.expect(api_get_survey_author)
+    def post(self):
+        if user_functions.get_user(api.payload['author']) is None:
+            return {'error_messages': "User not found"}
+        return survey_functions.author_get_survey(api.payload['author'])
+
+
 @api.route('/answer_a_survey')
 class answer_survey(Resource):
     @api.marshal_with(api_return_message)
     @api.expect(api_get_answerlist)
+    @login_required(role='ANY')
     def post(self):
         try:
             if survey_functions.id_get_survey(api.payload['survey_id']):
-                survey_functions.new_answerlist(api.payload)
+                now = survey_functions.id_get_survey(api.payload['survey_id'])
```

Comment:
`now` seems to be a timestamp, but I suppose it's a survey here. Could you change its name?

Reply:
change with tmp_survey